src/parachain/vaults.ts

```diff
@@ -346,6 +357,32 @@ export class DefaultVaultsAPI implements VaultsAPI {
         return this.wrappedCurrency;
     }
 
+    async registerNewCollateralVault(collateralAmount: MonetaryAmount<CollateralCurrencyExt>): Promise<void> {
+        // check the vault account is set
+        const vaultAccount = this.transactionAPI.getAccount();
+        if (vaultAccount === undefined) {
+            return Promise.reject("Vault account must be set in the vaults API");
```

Comment:
Technically the account is set globally in the `interbtcapi` instance (`transactionAPI` is shared by reference in all the APIs), so this error message could be misleading

Reply:
> Technically the account is set globally in the interbtcapi instance (transactionAPI is shared by reference in all the APIs), so this error message could be misleading

Good catch.
I confused myself there and ended up rewriting the message incorrectly. 
I would like to indicate that the error originated inside vaults API where it couldn't be retrieved.

I'll add that to my refactoring work (a collection of non-blocking PR feedback items)